lib/CommandCall.js

```diff
@@ -53,32 +44,32 @@ class CommandCall {
    * @returns {string}
    */
   toXML() {
-    if (this.type === 'sh') {
-      if (this.options) {
-        this.xml = iXmlNodeShOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error)
-              + this.command + iXmlNodeShClose();
-      } else {
-        this.xml = iXmlNodeShOpen('', '', '', '', '') + this.command + iXmlNodeShClose();
-      }
-    } else if (this.type === 'cl') {
-      const exec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
+    // consider renaming type: 'cl' -> cmd to avoid this check.
+    if (this.type === 'cl') {
+      this.type = 'cmd';
+    }
 
-      if (this.options) {
-        this.xml = iXmlNodeCmdOpen(this.options.exec, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeCmdClose();
-      } else {
-        this.xml = iXmlNodeCmdOpen(exec, '', '', '', '', '') + this.command + iXmlNodeCmdClose();
-      }
-    } else if (this.type === 'qsh') {
-      if (this.options) {
-        this.xml = iXmlNodeQshOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeQshClose();
-      } else {
-        this.xml = iXmlNodeQshOpen('', '', '', '', '') + this.command + iXmlNodeQshClose();
-      }
+    this.xml = `<${this.type}`;
```

Comment:
There's no checking if the type is valid. If the user passes in `type='mycommand'`, the behavior has changed.

Reply:
We do have a check for `type` in the [constructor](https://github.com/IBM/nodejs-itoolkit/blob/master/lib/CommandCall.js#L41).

Although unlikely the user could have adjusted the type when before calling `toXML`

Reply:
Oh, I missed that. It was probably fine as-is then, but better to be defensive.